### PR TITLE
feat: workflow docs

### PR DESCRIPTION
**Pull request by jnikhila**
## Description 

Provide a concise summary of the changes made in this pull request.

- A new section for Workflows added to the sidebar

    - Workflows
      - [Overview](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/workflows)
            -  A brief about workflows, components, and links to get started.
      - Tutorial
            - [Create a basic workflow](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/workflows/tutorials/create-workflow)
                   - Trigger it from the Appsmith app
                   - Integrate on Appsmith events and execute the workflow
                   - Added an interactive Arcade so that users can learn in an interactive mode

      - How-To guides
            -  [Create Approval Workflow](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/workflows/how-to-guides/create-approval-workflow)
                   - How to create an approval workflow that showcases a refund approval example
                   - Create a workflow to handle automated approvals and manual approval/rejections
                   - Trigger workflow to add new refund requests to the workflow
                   -  Read the pending refund requests in the Appsmith app
                   - Take action in the Appsmith app
                   - Resolve requests in Workflow
                   - Notify users of the outcome
       - Reference
              - [Workflow Queries](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/workflows/reference/workflow-queries)
                   - To give information on the type of workflow queries and their parameters 
              - [Workflow Functions](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/workflows/reference/workflow-functions)
      -  [Concept](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/workflows/concepts/workflows-in-appsmith)
                   -  To provide information on the Workflow feature in Appsmith and its key capabilities.
       - Granular Access Control
               - [ Roles >> Application resources](https://appsmith-docs-git-feat-workflow-tutorial-get-appsmith.vercel.app/advanced-concepts/granular-access-control/roles#application-resources)
                   - Updates to add Workflow permissions 
             
## Type of PR

Check the appropriate box:

- [ ] Review Fixes
- [ ] Documentation Overhaul
- [x] Feature/Story
    - Link one or more Engineering Tickets

        * 

- [ ] A-Force
- [ ] Error in documentation

## Documentation tickets
 Link to one or more documentation tickets:
 
 - #2076 
 - #2118 
 - #2100 
 - #2099 

## Checklist
Choose only the ones that are applicable.

- [x] Checked for Grammarly suggestions.
- [x] Adhered to the writing checklist.
- [x] Adhered the media checklist.
- [ ] Verified and updated cross-references or added redirect rules.
- [ ] Tested the redirect rules on deploy preview.
- [ ] Validated the modifications made to the content on deploy preview.
- [ ] Validated the CSS modifications on different screen sizes.

### CODE REVIEW

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/workflows/tutorials/set-up-webhook-workflow.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Follow these steps to create a new workflow within your workspace. The newly created workflow will be accessible across all apps in the same workspace:
+
+<div style={{ position: "relative", paddingBottom: "calc(50.520833333333336% + 41px)", height: "0", width: "100%" }}>
+  <iframe src="https://demo.arcade.software/TwBt2bvGNABi1Q0yZLS8?embed" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen style={{ position: "absolute", top: "0", left: "0", width: "100%", height: "100%", colorScheme: "light" }} title="Appsmith | Connect Data">
+  </iframe>
+</div>
+<br/><br/>
+
+1. Click the **Create New** button in your workspace, and from the available options, choose **New Workflow**. This action creates a new workflow in your workspace and takes you to the **Main** JS object code editor.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/tutorials/set-up-webhook-workflow.md:31]: I'll rewrite this.

File: website/docs/workflows/tutorials/set-up-webhook-workflow.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+```javascript
+export default {
+    async executeWorkflow(data) {
+        console.log("Workflow triggered");
+    }
+}
+```
+4. Click the gear icon ⚙️ in the bottom left corner to configure the workflow settings.
+5. Toggle the **Webhook trigger** property to configure the workflow as a webhook.
+6. Copy and save the **URL** and the **Bearer Token**.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/tutorials/set-up-webhook-workflow.md:44]: I'll check this.

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/workflows/reference/workflow-functions.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Workflow functions in Appsmith enable the creation of dynamic and interactive workflows, facilitating data manipulation, process automation, and user intervention within applications. This page provides information about the workflow functions available in Appsmith, including their signatures, parameters, and usage examples.
+
+## assignRequest()
+
+The `assignRequest()` function creates a pending request that requires user intervention and can be accessed later in your apps to enable users to take action.
+
+### Signature
+
+```javascript
+assignRequest({
```
Comment (jnikhila) [inline website/docs/workflows/reference/workflow-functions.md:13]: @infinitetrooper and @srix - what will be the return type for this?

File: website/docs/advanced-concepts/granular-access-control/roles.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ In this section, you'll find permissions related to the application and its reso
 | **App** | Create pages and queries inside the app | Edit pages and queries inside the app. | Delete the app and its pages and queries. | View the app and its pages and queries. | - | Make the application public | Export that particular application. |
 | **Page** | Create queries on the page. | Edit the page and its queries. | Delete the page and its queries. | View the page and its queries. | - | - | - |
 | **Query** | - | Edit the query. | Delete the query. | View the query. | Execute the query. | - | - |
-
+| **Workflows** | Create new workflows and resources inside a workflow. | Required to make changes to the Workflow.| Required to delete a workflow. Users will need **Edit** permission to delete individual actions inside the workflow. | An **Edit** permission also enables the permission to view workflow run history.| An **Edit** permission also enables the **Execute** permission to run workflow.| - | -|
```
Comment (jnikhila) [inline website/docs/advanced-concepts/granular-access-control/roles.md:44]: @infinitetrooper - Could you please confirm about the permissions for Publishing workflows?

**Review by infinitetrooper — CHANGES_REQUESTED** [review]

File: website/docs/workflows/reference/workflow-functions.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    ```
+  In your workflow, you can access properties within the `data` object like `userId` using dot notation. To access the `userId`, use `data.userId`.
+</dd>
+
+### Return type
+
+The `executeWorkflow()` returns a Promise that resolves to a boolean value, either `true` or `false`, indicating the success or failure of the workflow execution.
+
+## assignRequest()
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/reference/workflow-functions.md:60]: Here we are just talking about the function. So the title will be `assignRequest`. I can add in the description that it is available to be accessed using `appsmith.worfklow` global object available in workflows.

File: website/docs/workflows/reference/workflow-functions.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+The `assignRequest()` function creates a pending request that requires user intervention and can be accessed later in your apps to enable users to take action.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/reference/workflow-functions.md:62]: Rewritten this part to capture the necessary details.

File: website/docs/workflows/reference/workflow-functions.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    </dd>
+
+#### metadata `JSON` `Optional `
+    <dd>
+    Add data that may be needed to process the request or display more information to the user in your app. For example, you can include a unique identifier for the record associated with the request. Use the identifier in your app to fetch and show the details to user.
+    </dd>
+
+#### requestToUsers  
+   <dd>
+   Specifies the user or users to whom the request will be assigned for resolution. Each element in the array has to be the email that you use for logging into your Appsmith app. It's mandatory to supply this attribute if you are not supplying `requestToGroups` atribute.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/reference/workflow-functions.md:97]: It is mentioned that each element should be an email. However, have rewritten it.

File: website/docs/workflows/reference/workflow-functions.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+   </dd>
+
+#### requestToGroups `String[]` `Optional`
+
+<dd>
+Specifies the group name or names to which the request will be assigned for resolution. When specified, the request will be assigned to all the users belonging to the groups. Each group name must be configured in your app and have appropriate permissions to perform actions. It's mandatory to supply this attribute if you are not supplying the `requestToUsers` attribute.
+ </dd>
+
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/reference/workflow-functions.md:105]: what will be the impact of that when the workflow is unpaused?

File: website/docs/advanced-concepts/granular-access-control/roles.md (hunk 4 above)
Comment (jnikhila) [reply website/docs/advanced-concepts/granular-access-control/roles.md:44]: @infinitetrooper - did you get a chance to look into this? Please share what's missing here.

File: website/docs/workflows/tutorials/create-webhook-workflow.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
+---
+description: This page provides detailed steps to set up a Webhook workflow on Appsmith.
+title:  Create Webhook Workflow
+hide_title: true
+---
+<!-- vale off -->
+
+<div className="tag-wrapper">
+ <h1> Create Webhook Workflow</h1>
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/workflows/tutorials/create-webhook-workflow.md]: This is updated to `Create Basic Workflow.`

**Pull request merged**